packages/server-core/src/projects/scene/scene.hooks.ts

```diff
@@ -25,24 +25,275 @@ Ethereal Engine. All Rights Reserved.
+  const data = context.data as SceneCreateData
+  try {
+    const { name, sceneData, thumbnailBuffer } = data
+    let parsedSceneData = sceneData
+    if (sceneData && typeof sceneData === 'string') parsedSceneData = JSON.parse(sceneData)
+
+    logger.info('[scene.update]: ', name, data)
+
+    const storageProvider = getStorageProvider(data.storageProvider)
+
+    const newSceneJsonPath = `scenes/${name}/${name}.scene.json`
+    await storageProvider.putObject({
+      Key: newSceneJsonPath,
+      Body: Buffer.from(
+        JSON.stringify(cleanStorageProviderURLs(parsedSceneData ?? (defaultSceneSeed as unknown as SceneJsonType)))
+      ),
+      ContentType: 'application/json'
+    })
+
+    if (thumbnailBuffer && Buffer.isBuffer(thumbnailBuffer)) {
+      const sceneThumbnailPath = `scenes/${name}/${name}.thumbnail.ktx2`
+      await storageProvider.putObject({
+        Key: sceneThumbnailPath,
+        Body: thumbnailBuffer as Buffer,
+        ContentType: 'image/ktx2'
+      })
+    }
+
+    try {
+      await storageProvider.createInvalidation(sceneAssetFiles.map((asset) => `scenes/${name}/${name}${asset}`))
+    } catch (e) {
+      logger.error(e)
+      logger.info(sceneAssetFiles)
+    }
+  } catch (err) {
+    logger.error(err)
+    throw err
+  }
+}
+
+const getScene = async (context: HookContext<SceneService>) => {
+  context.result = await context.service._get(context.id as SceneID)
+}
+
+const deleteSceneResources = async (context: HookContext<SceneService>) => {
+  const sceneName = context.params?.query?.name
+  const storageProviderName = context.params?.query?.storageProvider
+  const storageProvider = getStorageProvider(storageProviderName)
+  const scenePath = `scenes/${sceneName}/${sceneName}.scene.json`
+  const thumbnailPath = `scenes/${sceneName}/${sceneName}.thumbnail.ktx2`
+  const envMapPath = `scenes/${sceneName}/${sceneName}.envmap.ktx2`
+
+  await storageProvider.deleteResources([scenePath, thumbnailPath, envMapPath])
+
+  try {
+    await storageProvider.createInvalidation([scenePath, thumbnailPath, envMapPath])
+  } catch (e) {
+    logger.error(e)
+    logger.info(sceneAssetFiles)
+  }
+  delete context.params.query?.storageProvider
+}
 
 export default {
   before: {
     all: [],
-    find: [],
-    get: [],
-    create: [iff(isProvider('external'), verifyScope('editor', 'write') as any, projectPermissionAuthenticate(false))],
-    update: [iff(isProvider('external'), verifyScope('editor', 'write') as any, projectPermissionAuthenticate(false))],
-    patch: [iff(isProvider('external'), verifyScope('editor', 'write') as any, projectPermissionAuthenticate(false))],
-    remove: [iff(isProvider('external'), verifyScope('editor', 'write') as any, projectPermissionAuthenticate(false))]
+    find: [persistPaginate],
+    get: [getSceneDataById],
+    create: [
+      iff(isProvider('external'), verifyScope('editor', 'write') as any, projectPermissionAuthenticate(false)),
```

Comment:
Suggestion:
`verifyScope('editor', 'write')`
_`as any`_ would not be required anymore